apps/extension/src/background/keyring/keyring.ts

```diff
@@ -121,11 +122,15 @@ export class KeyRing {
     await this.activeAccountStore.set(PARENT_ACCOUNT_ID_KEY, parentId);
   }
 
-  public async checkPassword(password: string): Promise<boolean> {
-    const activeAccountId = await this.getActiveAccountId();
+  public async checkPassword(password: string, accountId?: string): Promise<boolean> {
+    // default to active account if no account provided
+    let idToCheck = accountId;
```

Comment:
You could do this :)
`const idToCheck = accountId ?? (await this.getActiveAccountId());`

Reply:
I like that. It's much nicer. I'll add it in.